extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/MtlsAuthenticationMechanism.java

```diff
@@ -83,6 +83,11 @@ public Uni<HttpCredentialTransport> getCredentialTransport(RoutingContext contex
         return Uni.createFrom().item(new HttpCredentialTransport(HttpCredentialTransport.Type.X509, "X509"));
     }
 
+    @Override
```

Comment:
I think if we have inclusive authentication, then `HttpAuthenticator` must throw ISE if, after sorting enabled mechanisms, MTLS is in the list of the mechanisms but is not the first one. It has to be predictable IMHO, otherwise for ex, #4482 won't work. MTLS is the closest to the transport, so it won't be wrong to insist on it coming up first in case of the inclusive authentication. I'm on;y proposing to have this extra check, even though `3000` should guarantee it being first in most cases

Reply:
Okay, I think it sounds reasonable considering, as you said, is actually enabled and present. I'll add it.

Reply:
Added validation and test.